lib/internal/modules/cjs/loader.js

```diff
@@ -286,12 +286,12 @@ function readPackageScope(checkPath) {
   const rootSeparatorIndex = checkPath.indexOf(path.sep);
   let separatorIndex;
   while (
-    (separatorIndex = checkPath.lastIndexOf(path.sep)) > rootSeparatorIndex
+    (separatorIndex = checkPath.lastIndexOf(path.sep)) >= rootSeparatorIndex
```

Comment:
```suggestion
    (separatorIndex = StringPrototypeLastIndexOf(checkPath, path.sep)) >= rootSeparatorIndex
```
(i know this predates this PR, but might as well clean it up while you're in here)